### PR TITLE
[Metricbeat] Rename tags to tags_filter for cloudwatch metricset

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
This PR is to deprecate `tags` in cloudwatch metricset config. This is replaced by `tags_filter`, which is a metricset level config parameter for all metricsets under aws module.

### CODE REVIEW

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -305,22 +315,33 @@ func (m *MetricSet) readCloudwatchConfig() (listMetricWithDetail, map[string][]n
 
 			if config.ResourceTypeFilter != "" {
 				if _, ok := resourceTypesWithTags[config.ResourceTypeFilter]; ok {
-					resourceTypesWithTags[config.ResourceTypeFilter] = config.Tags
-
+					if config.TagsFilter != nil {
```
Comment (exekias) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go]: WDYT about doing this logic once at the beginning of this function and then use a variable for tags in all the subsequent uses?

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -66,7 +67,16 @@ type Config struct {
 	Dimensions         []Dimension `config:"dimensions"`
 	ResourceTypeFilter string      `config:"tags.resource_type_filter"`
 	Statistic          []string    `config:"statistic"`
-	Tags               []aws.Tag   `config:"tags"`
+	TagsFilter         []aws.Tag   `config:"tags_filter"`
+	Tags               []aws.Tag   `config:"tags"` // Deprecated.
```
Comment (exekias) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go]: +1 to doing this change! great work at keeping backwards compatibility

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go]: Good point!

**Comment by kaiyan-sheng** [comment]
Will come back to work on this after https://github.com/elastic/beats/pull/16605 is merged. In https://github.com/elastic/beats/pull/16605 I moved `tags_filter` to a MetricSet level config for aws module, which means all metricset under aws module should be able to use `tags_filter`. This means I have to remove `tags` under cloudwatch and instead to use `tags_filter`.

**Pull request merged**